scripts/exregional_run_met_ascii2nc_obs.sh

```diff
@@ -0,0 +1,376 @@
+#
+LEADHR_LIST=""
+num_missing_files=0
+for yyyymmddhh in ${obs_retrieve_times_crnt_day[@]}; do
+  yyyymmdd=$(echo ${yyyymmddhh} | cut -c1-8)
+  hh=$(echo ${yyyymmddhh} | cut -c9-10)
+
+  # Set the full path to the final processed obs file (fp_proc) we want to
+  # create.
+  sec_since_ref=$(${DATE_UTIL} --date "${yyyymmdd} ${hh} hours" +%s)
+  lhr=$(( (sec_since_ref - sec_since_ref_task)/3600 ))
+
+  fp=$( python3 $USHdir/eval_metplus_timestr_tmpl.py \
+    --init_time="${yyyymmdd_task}00" \
+    --lhr="${lhr}" \
+    --fn_template="${OBS_DIR}/${OBS_INPUT_FN_TEMPLATE}") || \
+    print_err_msg_exit "Call to eval_metplus_timestr_tmpl.py failed with return code: $?"
+
+  if [[ -f "${fp}" ]]; then
+    print_info_msg "
+Found ${OBTYPE} obs file corresponding to observation retrieval time (yyyymmddhh):
+  yyyymmddhh = \"${yyyymmddhh}\"
+  fp = \"${fp}\"
+"
+    hh_noZero=$((10#${hh}))
+    LEADHR_LIST="${LEADHR_LIST},${hh_noZero}"
+  else
+    num_missing_files=$((num_missing_files+1))
+    print_info_msg "
+${OBTYPE} obs file corresponding to observation retrieval time (yyyymmddhh)
+does not exist on disk:
+  yyyymmddhh = \"${yyyymmddhh}\"
+  fp = \"${fp}\"
+Removing this time from the list of times to be processed by ${METPLUSTOOLNAME}.
+"
+  fi
+done
+
+# If the number of missing files is greater than the maximum allowed
+# (specified by num_missing_files_max), print out an error message and
+# exit.
+if [ "${num_missing_files}" -gt "${NUM_MISSING_OBS_FILES_MAX}" ]; then
+  print_err_msg_exit "\
+The number of missing ${OBTYPE} obs files (num_missing_files) is greater
+than the maximum allowed number (NUM_MISSING_FILES_MAX):
+  num_missing_files = ${num_missing_files}
+  NUM_MISSING_OBS_FILES_MAX = ${NUM_MISSING_OBS_FILES_MAX}"
+fi
+
+# Remove leading comma from LEADHR_LIST.
+LEADHR_LIST=$( echo "${LEADHR_LIST}" | $SED "s/^,//g" )
+print_info_msg "$VERBOSE" "\
+Final (i.e. after filtering for missing obs files) set of lead hours
+(saved in a scalar string variable) is:
+  LEADHR_LIST = \"${LEADHR_LIST}\"
+"
+#
+#-----------------------------------------------------------------------
+#
+# Make sure the MET/METplus output directory(ies) exists.
+#
+#-----------------------------------------------------------------------
+#
+mkdir -p "${OUTPUT_DIR}"
+#
+#-----------------------------------------------------------------------
+#
+# Check for existence of top-level OBS_DIR.
+#
+#-----------------------------------------------------------------------
+#
+if [ ! -d "${OBS_DIR}" ]; then
+  print_err_msg_exit "\
+OBS_DIR does not exist or is not a directory:
+  OBS_DIR = \"${OBS_DIR}\""
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Export variables needed in the common METplus configuration file (at
+# ${METPLUS_CONF}/common.conf).
+#
+#-----------------------------------------------------------------------
+#
+export METPLUS_CONF
+export LOGDIR
```

Comment:
Do these have values? Where are those set prior to this point in the script? Why do they need to be exported? What uses them downstream?

Reply:
`LOGDIR` is set at the rocoto task level, and `METPLUS_CONF` comes from the config file. These variables are needed because they are referenced in the main METplus `common.conf` file.